webpack/assets/javascripts/react_app/components/ForemanModal/ForemanModal.js

```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+import { Modal } from 'patternfly-react';
+import PropTypes from 'prop-types';
+import ForemanModalHeader from './subcomponents/ForemanModalHeader';
+import ForemanModalFooter from './subcomponents/ForemanModalFooter';
+import ModalContext from './ForemanModalContext';
+import { extractModalNodes } from './helpers';
+
+const ForemanModal = props => {
+  const { isOpen, onClose, title, ...propsToPassDown } = props; // don't pass down those two as it causes PF problems
+  // Extract header and footer from children, if provided
+  const { headerChild, footerChild, otherChildren } = extractModalNodes(
+    props.children
+  );
+  const context = {
+    isOpen,
+    onClose,
+    title,
+  };
+
+  return (
+    <ModalContext.Provider value={context}>
```

Comment:
I'm not sure if React's context is the right thing here
according to react's docs _"Context is designed to share data that can be considered “global” for a tree of React components , such as the current authenticated user, theme, or preferred language."_ while here the context is used only for a very specific use - modals props, which are not a global data nor tree of react components.

Reply:
@amirfefer I feel like I want to leave it for now because I feel it will make the compound component more scalable in the future.  See the discussion from above with Avi: https://github.com/theforeman/foreman/pull/7039#discussion_r328287396

Reply:
I missed that comment, thanks!
That blog post is really interesting and shows a nice flow, I wonder why React docs are more strict about using context, but I do think it looks much cleaner